build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,484 @@
+
+                dir('package.tests') {
+                    deleteDir()
+                }
+
+                dir('xamarin-android@tmp') {
+                    deleteDir()
+                }
+
+                dir ('xamarin-android/package-test-errors') {
+                    deleteDir()
+                }
+
+                dir('xamarin-android') {
+                    script {
+                        echo "Cleaning build outputs: xamarin.android-oss*.zip, bin/${env.BuildFlavor}/bundle-*.zip, bin/Build*/Xamarin.Android.Sdk*.vsix"
+                        cleanFiles("xamarin.android-oss*.zip")
+                        cleanFiles("bin/${env.BuildFlavor}/bundle-*.zip")
+                        cleanFiles("bin/Build*/Xamarin.Android.Sdk*.vsix")
+
+                        echo "Cleaning xa-build-status*"
+                        cleanFiles("xa-build-status*.zip")
+                        cleanDirectories("xa-build-status*")
+                        cleanFiles("build-status*")
+
+                        echo "Cleaning xa-test-errors*"
+                        cleanFiles("xa-test-errors*.zip")
+                        cleanDirectories("xa-test-errors*")
+
+                        echo "Cleaning prepare-image-dependencies.sh"
+                        sh "rm -f prepare-image-dependencies.sh"
+
+                        echo "Cleaning test result files 'TestResult-*.xml'"
+                        cleanFiles("TestResult-*.xml")
+
+                        // Avoid: EXEC : error : jnimarshalmethod-gen: Unable to create Java VM
+                        def XAInstallPrefix = "bin/${env.BuildFlavor}/lib/xamarin.android"
+
+                        def javaInteropDllConfigFile = "${XAInstallPrefix}/xbuild/Xamarin/Android/Java.Interop.dll.config"
+                        if (fileExists(javaInteropDllConfigFile)) {
+                            echo "Cleansing file '${javaInteropDllConfigFile}'"
+                            sh "rm -f ${javaInteropDllConfigFile}"
+                        }
+
+                        def javaEnvironmentConfigFile = "${XAInstallPrefix}/xbuild/Xamarin/Android/Java.Runtime.Environment.dll.config"
+                        if (fileExists(javaEnvironmentConfigFile)) {
+                            echo "Cleansing file '${javaEnvironmentConfigFile}'"
+                            sh "rm -f ${javaEnvironmentConfigFile}"
+                        }
+                    }
+                }
+
+                echo "Stage result: clean: ${currentBuild.currentResult}"
+            }   
+        }
+    
+        stage('prepare deps') {
+            options {
+                timeout(time: 30, unit: 'MINUTES')  // Typically takes less than 2 minutes
+            }
+
+            steps {
+                dir("$XADir") {
+                    sh "make prepare-deps CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                }
+
+                echo "Stage result: prepare deps: ${currentBuild.currentResult}"
```

Comment:
What is `currentBuild.currentResult`?  The "exit value" of the last `sh` block?  Something else?

Do we need to print out this message?

Alternatively, does/will `sh` *quote*/echo the command it's executing as part of the execution?

Reply:
Yes, this message has proven useful in a couple ways. a) determining from the log which stage led to the build being marked as UNSTABLE (as it is not always clear from the Jenkins pipeline UI);  b). since any statement--not just sh calls--within the stage could lead to failures (as interpreted by Jenkins), the status provides the failure assessment as Jenkins sees it (and will in turn be reported back).  BTW: I also tend to now search for the 'Stage result:' tag in the log to quickly assess the result for each stage.